### PR TITLE
[SPARK-27161][SQL] improve the document of SQL keywords

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

Make it more clear about how Spark categories keywords regarding to the config `spark.sql.parser.ansi.enabled`

## How was this patch tested?

existing tests

### CODE REVIEW

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -1215,6 +1232,9 @@ nonReserved
     | YEARS
     ;
 
+//============================
+// Start of the keywords list
+//============================
 SELECT: 'SELECT';
```
Comment (cloud-fan) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: We should sort the keyword list. It's better to do it in a followup to keep the diff small for this PR.

File: docs/sql-keywords.md [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 
-The list of reserved and non-reserved keywords can change according to the config
-`spark.sql.parser.ansi.enabled`, which is false by default.
+When `spark.sql.parser.ansi.enabled` is false, Spark SQL has two kinds of keywords:
+* Non-reserved keywords: Keywords that have a special meaning only in particular contexts and can be used as identifiers in other contexts.
+* Strict-non-reserved keywords: A strict version of non-reserved keywords, which can not be used as table alias.
+
+By default `spark.sql.parser.ansi.enabled` is false.
+
+Below is a list of all the keywords in Spark SQL.
```
Comment (cloud-fan) [inline docs/sql-keywords.md]: I noticed that this list does not exactly match the keywords in Spark, e.g. `ABS` is not keyword in Spark, `SETMINUTE` is not in this list.

Comment (cloud-fan) [reply docs/sql-keywords.md]: @maropu can you take a closer look later?

**Comment by cloud-fan** [comment]
cc @gatorsmile

File: docs/sql-keywords.md (hunk 2 above)
Comment (maropu) [reply docs/sql-keywords.md]: ok, I'll check and fix as followup.

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 (hunk 1 above)
Comment (maropu) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: +1

File: docs/sql-keywords.md [hunk 3]
```diff
@@ -1,16 +1,20 @@
 ---
 layout: global
-title: SQL Reserved/Non-Reserved Keywords
-displayTitle: SQL Reserved/Non-Reserved Keywords
+title: Spark SQL Keywords
+displayTitle: Spark SQL Keywords
```
Comment (maropu) [inline docs/sql-keywords.md]: `spark.sql.parser.ansi.enabled` affects parsing behaviours, too, e.g., when true, it makes `interval` optional. In future, we could change the behaivour of overflow handling in execution for the more strict ANSI compliance. These behaivour changes affected by the ANSI option should be documented not in this document but in another document?

File: docs/sql-keywords.md [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
-table alias, but can be used as other identifiers.
+When `spark.sql.parser.ansi.enabled` is true, Spark SQL has two kinds of keywords:
+* Reserved keywords: Keywords that reserved and can't be used as identifiers for table, view, column, alias, etc.
+* Non-reserved keywords: Keywords that have a special meaning only in particular contexts and can be used as identifiers in other contexts.
 
-The list of reserved and non-reserved keywords can change according to the config
-`spark.sql.parser.ansi.enabled`, which is false by default.
+When `spark.sql.parser.ansi.enabled` is false, Spark SQL has two kinds of keywords:
+* Non-reserved keywords: Keywords that have a special meaning only in particular contexts and can be used as identifiers in other contexts.
+* Strict-non-reserved keywords: A strict version of non-reserved keywords, which can not be used as table alias.
```
Comment (maropu) [inline docs/sql-keywords.md]: Great and this new group is easy-to-understand.

File: docs/sql-keywords.md (hunk 3 above)
Comment (cloud-fan) [reply docs/sql-keywords.md]: Yea, this document is about keywords, not everything about the ansi mode.

**Review by maropu — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
retest this please

**Comment by maropu** [comment]
retest this please

**Comment by maropu** [comment]
Thanks! Merged to master

**Comment by maropu** [comment]
I'll fix some minor issues in follow-up.

**Pull request closed without merging**